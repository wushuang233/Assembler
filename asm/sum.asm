# 计算从1到10的累加和
# R1 - 累加和
# R2 - 终止值 (10)
# R3 - 当前值 (循环变量)

# 初始化
addi x1, x0, 0    # R1 = 0 (累加和初始化为0)
addi x2, x0, 10   # R2 = 10 (终止值)
addi x3, x0, 0    # R3 = 0 (循环变量初始化为0)

# 循环
# R3 = R3 + 1 (增加循环变量)
addi x3, x3, 1
# R1 = R1 + R3 (累加)
add x1, x1, x3
# 如果R3 != R2，则继续循环
bne x3, x2, -8

# 程序结束，R1中保存累加和 (应为55)
halt 