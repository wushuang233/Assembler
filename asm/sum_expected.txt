# 计算从1到10的累加和的预期二进制输出
# addi x1, x0, 0 (R1 = 0)
0b00000000000_00000_00000_00001_000010
# addi x2, x0, 10 (R2 = 10)
0b00000000000_01010_00000_00010_000010
# addi x3, x0, 0 (R3 = 0)
0b00000000000_00000_00000_00011_000010
# addi x3, x3, 1 (R3 = R3 + 1)
0b00000000000_00001_00011_00011_000010
# add x1, x1, x3 (R1 = R1 + R3)
0b00000000000_00011_00001_00001_000001
# bne x3, x2, -8 (如果R3 != R2，跳回8字节)
0b11111111111_00010_00011_11000_000011
# halt
0b00000000000_00000_00000_00000_000000 